samples/ask-akka-agent/src/main/java/akka/ask/common/KeyUtils.java

```diff
@@ -1,18 +1,32 @@
 package akka.ask.common;
 
+import com.typesafe.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class KeyUtils {
 
+  private static final Logger logger = LoggerFactory.getLogger(KeyUtils.class);
+
   public static String readMongoDbUri() {
     return System.getenv("MONGODB_ATLAS_URI");
   }
 
-  public static String readOpenAiKey() {
-    return System.getenv("OPENAI_API_KEY");
+  public static void checkKeys(Config config) {
+    if (config.getString("akka.javasdk.agent.model-provider").equals("openai")
```

Comment:
Initially, I thought of moving that logging to the SDK, as @pvlugter suggested, but I'm not sure about that anymore. Maybe it's better to leave that for users, where they can decide whether this should be a warning or error, and what the message should be.

Reply:
Mentioned in another comment: could we instead add some kind of check method to `ModelProvider` and then each model provider implementation can have checks for its own settings. If the user code can then access the currently configured model provider, and call the general check, and can then decide how to report.

Reply:
Yeah, but a simple check like `boolean isValid()` is also not perfect, because the API key might be present, but the model name is empty. On the other hand, a check like `Validation isValid()` that would also carry the information of the error seems like over-engineering. WDYT?